pkg/application/applyBlueprintSpecUseCase.go

```diff
@@ -0,0 +1,145 @@
+		WithValues("blueprintId", blueprintId)
+
+	logger.Info("getting blueprint spec for checking ecosystem health afterwards")
+	blueprintSpec, err := useCase.repo.GetById(ctx, blueprintId)
+	if err != nil {
+		return fmt.Errorf("cannot load blueprint spec %q to check ecosystem health: %w", blueprintId, err)
+	}
+
+	healthResult, err := useCase.healthUseCase.WaitForHealthyEcosystem(ctx)
+	if err != nil {
+		return fmt.Errorf("cannot check ecosystem health after applying the blueprint %q: %w", blueprintId, err)
+	}
+	blueprintSpec.CheckEcosystemHealthAfterwards(healthResult)
+
+	err = useCase.repo.Update(ctx, blueprintSpec)
+	if err != nil {
+		return fmt.Errorf("cannot save blueprint spec %q after checking the ecosystem health: %w", blueprintId, err)
+	}
+
+	return nil
+}
+
+// TODO: activate maintenance mode
+func (useCase *ApplyBlueprintSpecUseCase) ApplyBlueprintSpec(ctx context.Context, blueprintId string) error {
+	blueprintSpec, err := useCase.repo.GetById(ctx, blueprintId)
+	if err != nil {
+		return fmt.Errorf("cannot load blueprint to apply blueprint spec: %w", err)
+	}
+
+	err = useCase.markInProgress(ctx, blueprintSpec)
+	if err != nil {
+		return err
+	}
+
+	applyError := useCase.doguInstallUseCase.ApplyDoguStates(ctx, blueprintId)
+	if applyError != nil {
+		err := useCase.MarkFailed(ctx, blueprintSpec, err)
+		if err != nil {
+			return err
+		}
+		return applyError
+	}
+
+	return useCase.markBlueprintApplied(ctx, blueprintSpec)
+}
+
+//TODO: deactivate maintenance mode
+
+func (useCase *ApplyBlueprintSpecUseCase) markInProgress(ctx context.Context, blueprintSpec *domain.BlueprintSpec) error {
+	blueprintSpec.MarkInProgress()
+	err := useCase.repo.Update(ctx, blueprintSpec)
+	if err != nil {
+		return fmt.Errorf("cannot mark blueprint as in progress: %w", err)
+	}
+	return nil
+}
+
+func (useCase *ApplyBlueprintSpecUseCase) MarkFailed(ctx context.Context, blueprintSpec *domain.BlueprintSpec, err error) error {
+	logger := log.FromContext(ctx).
+		WithName("ApplyBlueprintSpecUseCase.MarkFailed").
+		WithValues("blueprintId", blueprintSpec.Id)
+
+	blueprintSpec.MarkFailed(err)
+	repoErr := useCase.repo.Update(ctx, blueprintSpec)
+
+	if repoErr != nil {
+		repoErr = errors.Join(repoErr, err)
+		logger.Error(repoErr, "cannot mark blueprint as failed")
+		return fmt.Errorf("cannot mark blueprint as failed while handling %q status: %w", blueprintSpec.Status, repoErr)
+	}
+	return nil
+}
+
+func (useCase *ApplyBlueprintSpecUseCase) markBlueprintApplied(ctx context.Context, blueprintSpec *domain.BlueprintSpec) error {
+	blueprintSpec.MarkBlueprintApplied()
+	err := useCase.repo.Update(ctx, blueprintSpec)
+	if err != nil {
+		return fmt.Errorf("cannot mark blueprint as waiting for a healthy ecosystem: %w", err)
+	}
+	return nil
+}
+
+func (useCase *ApplyBlueprintSpecUseCase) markCompleted(ctx context.Context, blueprintSpec *domain.BlueprintSpec) error {
```

Comment:
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>func `(*ApplyBlueprintSpecUseCase).markCompleted` is unused

Reply:
This will be used in the next PR for the maintenance mode, which will finally mark the blueprint as completed:
`healthyUpfront`->`maintenanceModeActivated`->`inProgress`->`stateApplied`->`healthyAfterwards`->`completed` (maintenanceModeDeactivated)